src/librbd/ImageWatcher.cc

```diff
@@ -1052,6 +1060,17 @@ void ImageWatcher<I>::handle_rewatch(int r) {
   CephContext *cct = m_image_ctx.cct;
   ldout(cct, 10) << this << " " << __func__ << ": r=" << r << dendl;
 
+  m_watch_error_count -= 1;
+  if (m_watch_error_count > 0) {
```

Comment:
Make sure you only attempt a re-watch if you don't have a pending de-register

Reply:
Ah, so we a de-register means we just stop at that point. I guess that warrants a small bit of refactoring here. I'll try that.